operator/v1/types_network.go

```diff
@@ -332,6 +336,14 @@ type HybridOverlayConfig struct {
 	HybridOverlayVXLANPort *uint32 `json:"hybridOverlayVXLANPort,omitempty"`
 }
 
+type IPsecConfig struct {
+	// enable enables IPsec encryption for pod-to-pod traffic on the pod network
+	// within the cluster. Default is false.
+	// +optional
+	// +kubebuilder:default:=false
+	Enable bool `json:"enable"`
```

Comment:
Sorry I know this is coming a bit late already, but does it make sense for the default to be true in case the IPsecConfig is present? I don't see that default false to be a natural UX i.e. when someone specifies an IPsecConfig, more often than not they want IPsec configured.

Reply:
Better yet, why do we even need the `Enable` flag? Isn't the presence/absence of `ipsecconfig` a good enough flag?

Reply:
Good point .. @squeed @sttts What do you think?

Reply:
the convention if a feature takes no configuration now, but will later, is to have an empty object be enabled, and `null` be disabled.

e.g.
```yaml
foo: bar
ipsec: {}
```

Reply:
you can also make `enable` required without a default. Then it must be true or false, i.e. explicitly specified.

Reply:
@squeed Ok, so that would mean removing the flag as-per Aniket's comment?

Reply:
I think I prefer @squeed 's suggestion personally. Just adding @danwinship as he also reviewed.

Reply:
either is fine with me. We have cases like empty dir volumes that work this way.

Reply:
It looks like there is consensus. I prefer having the empty struct `{} ` indicate the feature is enabled and `null` imply that ipsec is disabled.

Reply:
Updated and here is the PR @abhat @sttts : https://github.com/openshift/api/pull/794